nixos/modules/system/etc/etc-activation.nix

```diff
@@ -34,11 +34,28 @@
         mounts = [
           {
             where = "/run/etc-metadata";
-            what = "/sysroot${config.system.build.etcMetadataImage}";
+            what = "/tmp/etc-metadata-image";
```

Comment:
That doesn't actually work because we don't have `/nix` in initrd, so the symlink in the toplevel appears to be broken from inside the initrd. We need to chroot into `/sysroot` in order for the symlinks inside the nix store to work, which is why we have the service where we run realpath inside a chroot to find the right path in the nix store.

Reply:
The alternative would be to bind mount `/sysroot/nix` to `/nix` in the initrd, which will make the symlinks work correctly. In that case I think we don't need the service here anymore.

Reply:
Can't you just prepend `/sysroot` to the path? This is also what we do in the overlayfs abstraction.

Reply:
If we want to avoid the additional service, I think the best way would be to bind mount /sysroot/nix to /nix in initrd, in that case we can just have `what = /nixos-closure/etc-metadata-image` and the path can be correctly normalised. Does that sound like a good solution to you?
I will probably do that in a follow-up PR then, to avoid more discussion here.

Reply:
Not sure if this is a good solution but we could discuss that in a follow up PR.